### ISSUE TITLE
Proposal: Looking for Maintainer Policy

### ISSUE THREAD

**Issue by Justinzobel** [getsolus/packages#1443]
I'd like to propose a deprecation policy for packages that lose their maintainer.

Example:
Bob is the maintainer of packages foo, bar and baz. Bob moves on from Solus and now those packages have no maintainer.

Packages foo, bar and baz are placed in a shared Solus spreadsheet to track.

Each month on a certain day/date a post is made in the forums with a list of packages looking for a maintainer:
```
Looking for maintainer:

foo
bar
baz

Spiel about how it's an easy way to help Solus, we will help you with the technical skills along the way, etc.
```

A package remains on the maintainer wanted list for 2 forum posts and on the date of the 3rd forum post, if it has not been "adopted", it is deprecated. On that forum post it is mentioned at the bottom that the below packages did not find a maintainer and have been dropped (was going to say deprecated but dropped is more user friendly) from the Solus repository.

**Comment by rav101** [comment]
Would the forum posts be used as a history of previously maintained (and subsequently dropped) packages? Where would the package.yml files live for dropped packages? Would be good if they could be kept somewhere so if someone wanted to pick up a dropped package say 6 months down the line, they have an easy first port of call to find the cheat sheet to update it to current?

I'm thinking this could be a great place for new people who want to learn to package?

**Comment by Justinzobel** [comment]
Yeah, I think having an archive of deprecated package.yml (and associated patches if relevant) is a good idea. They could be placed in a new repo on GitHub called package-archive. README should clearly state that these aren't in the repo and were deprecated. If you'd like to maintain one of these packages, please place a new package request and make note that it was previously in the repo and link to the package.yml here.

**Comment by Justinzobel** [comment]
I've updated the name of this post as I'm thinking we can include packages that have been requested and approved but are looking for a maintainer, it's all the same situation, packages looking for a maintainer.

For packages that aren't in the repo yet but looking for maintainer, I would propose a longer timeframe around them, perhaps 1 year.

In the monthly post we would list these packages that are requested, approved and have the maintainer wanted label. Then it allows more users to see that "Oh XYZ package is requested, and I would like that in the repo too, I might maintain that".

**Comment by sebo505** [comment]
As a normal noob enduser a support this proposal.
While normal end-users might check the forums from time to time, it is highly unlikely for them to follow dev tracker or Matrix channels.
For those users deprecation comes with surprise and without warning.
I remember like 2 years ago that suddenly some of my scripts stopped working as expected due to a "silent" deprecation of yad.
Took some time for me to figure out what had happened and had then to compile yad on my own.

Deprecation of packages should also be announced in weekly update notes.
Or in a seperate thread per deprecated packages. That might be easier to find via search.

**Comment by sebo505** [comment]
Also nice would be a deprecation log file on your system that lists packages that got removed from your system due to deprecation.
And a notification on your screen triggered after update has finnished like "Update complete. The following packages got removed due to deprecation: x, y,z"

**Comment by ermo** [comment]
@davidjharder Assigning this to you so you can keep us on our toes.

**Comment by Justinzobel** [comment]
Ping.

**Comment by davidjharder** [comment]
Pong?
